packages/parcel-transformer/src/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { Transformer } from '@parcel/plugin';
+import compiledBabelPlugin from '@compiled/babel-plugin';
+import { parseAsync, transformFromAstAsync } from '@babel/core';
+import generate from '@babel/generator';
+
+/**
+ * Compiled parcel transformer.
+ */
+export default new Transformer({
+  canReuseAST() {
+    // Compiled should run before any other JS transformer.
+    return false;
+  },
+
+  async parse({ asset }: any) {
+    const code = await asset.getCode();
+    if (code.indexOf('@compiled/react') === -1) {
+      // We only want to parse files that are actually using Compiled.
+      // For everything else we bail out.
+      return undefined;
+    }
+
+    const ast = await parseAsync(code, {
+      filename: asset.filePath,
+      caller: { name: 'compiled' },
+      // TODO: We want to inherit user land configuration for the initial parse,
+      // but Parcel also ships with a "default" config. We should figure out if we
+      // can grab theirs.
+      plugins: ['@babel/plugin-syntax-jsx'],
+    });
+
+    return ast;
+  },
+
+  async transform({ asset, ast }: any) {
+    if (!asset.isSource || !ast) {
+      // We will only recieve asts for assets we're interested in.
+      // Since this is undefined we aren't interested in it.
+      return [asset];
+    }
+
+    const includedFiles: string[] = [];
+    const code = asset.isASTDirty() ? undefined : await asset.getCode();
+
+    const result = await transformFromAstAsync(ast.program, code, {
+      code: false,
+      ast: true,
+      filename: asset.filePath,
+      babelrc: false,
+      configFile: false,
+      plugins: [
+        [
+          compiledBabelPlugin,
+          { onIncludedFile: (file: string) => includedFiles.push(file), cache: false },
+        ],
+      ],
+      caller: {
+        name: 'compiled',
+      },
+    });
+
+    includedFiles.forEach((file) => {
+      asset.addIncludedFile(file);
+    });
+
+    if (result?.ast) {
+      asset.setAST({
+        // TODO: Currently if we set this as `'babel'` the babel transformer blows up.
+        // Let's figure out what we can do to reuse it.
+        type: 'compiled',
+        version: '0.0.0',
```

Comment:
This should be 
```
      type: 'babel',
      version: '7.0.0',
```

Reply:
Thanks! I had it like that before but it was blowing up. I'll try again after I upgrade the nightly and get everything working.

Reply:
@mischnic i get this error when updating this to babel - does the babel transformer not accept a standard babel AST?

```
$ parcel serve ./src/index.html --open
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
Error: this[statement.type] is not a function
TypeError: this[statement.type] is not a function
    at Object.Program (/Users/madou/projects/compiled/node_modules/astring/dist/astring.js:228:27)
    at Object.Program (/Users/madou/projects/compiled/node_modules/@parcel/babel-ast-utils/lib/generator.js:54:34)
    at Object.generator.<computed> [as Program] (/Users/madou/projects/compiled/node_modules/@parcel/babel-ast-utils/lib/generator.js:317:10)
    at generate (/Users/madou/projects/compiled/node_modules/astring/dist/astring.js:1149:29)
    at generateAST (/Users/madou/projects/compiled/node_modules/@parcel/babel-ast-utils/lib/index.js:123:43)
    at generate (/Users/madou/projects/compiled/node_modules/@parcel/babel-ast-utils/lib/index.js:163:10)
    at async /Users/madou/projects/compiled/node_modules/@parcel/core/lib/Transformation.js:456:24
    at async Promise.all (index 0)
    at async Transformation.runPipeline (/Users/madou/projects/compiled/node_modules/@parcel/core/lib/Transformation.js:452:7)
    at async Transformation.runPipelines (/Users/madou/projects/compiled/node_modules/@parcel/core/lib/Transformation.js:368:40)
✨  Done in 36.16s.
➜  compiled git:(parcel-plugin) ✗
```